### PR TITLE
[CIS-2013] Support channel truncation in Mock Server

### PR DESCRIPTION
**Pull request by alteral**
### 🔗 Issue Links

- https://stream-io.atlassian.net/browse/CIS-2013

### 🎯 Goal

- Support channel truncation in Mock Server
- Automate related scenarios

### 📝 Summary

- Mock server supports truncate channel feature
- Related scenarios are automated
- Some additional scenarios were automated for the company
- Truncate with & w/o buttons were added to DebugMenu in the TestApp
- Some refactoring of the mock server and the tests has been done

### ☑️ Contributor Checklist
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [ ] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [x] This change follows zero ⚠️ policy (required)
- [ ] Comparison screenshots added for visual changes
- [ ] Affected documentation updated (docusaurus, tutorial, CMS)

### 🎁 Meme

gif

### CODE REVIEW

File: StreamChatUITestsApp/DebugMenu.swift [hunk 1]
```diff
@@ -62,7 +62,33 @@ final class DebugMenu {
...
+                        .init(title: "Truncate channel w/o message",
+                              style: .default,
+                              handler: { _ in
+                                channelController.truncateChannel { [unowned self] error in
+                                    if let error = error {
+                                        self.presentAlert(in: viewController,
+                                                          title: "Couldn't truncate channel",
+                                                          message: "\(error.localizedDescription)",
+                                                          actions: []
+                                        )
```
Comment (nuno-vieira) [inline StreamChatUITestsApp/DebugMenu.swift:75]: To keep the code consistent, we should move all the parameters into a new line, to keep things tidier. This for formatting is the old Objc one and we should avoid it 👍 (At least in this codebase for consistency)
```suggestion
                                        self.presentAlert(
                                                 in: viewController,
                                                 title: "Couldn't truncate channel",
                                                 message: "\(error.localizedDescription)",
                                                 actions: []
                                        )
```

File: StreamChatUITestsApp/DebugMenu.swift [hunk 2]
```diff
@@ -62,7 +62,33 @@ final class DebugMenu {
...
+                        .init(title: "Truncate channel with message",
+                              style: .default,
+                              handler: { _ in
+                                channelController.truncateChannel(systemMessage: "Channel truncated") { [unowned self] error in
+                                    if let error = error {
+                                        self.presentAlert(in: viewController,
+                                                          title: "Couldn't truncate channel",
+                                                          message: "\(error.localizedDescription)",
+                                                          actions: []
+                                        )
```
Comment (nuno-vieira) [inline StreamChatUITestsApp/DebugMenu.swift:88]: Same here

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅ Just added some nit comments :)

File: StreamChatUITestsApp/DebugMenu.swift (hunk 1 above)
Comment (alteral) [reply StreamChatUITestsApp/DebugMenu.swift]: Makes sense 👍

File: StreamChatUITestsApp/DebugMenu.swift (hunk 2 above)
Comment (alteral) [reply StreamChatUITestsApp/DebugMenu.swift]: Done 👍

**Pull request merged**